client/v3/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,166 @@
+package kubernetes
+
+import (
+	"context"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	"go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// New creates KubernetesClient from config.
+// Caller is responsible to call Close() to clean up client.
+func New(cfg clientv3.Config) (*KubernetesClient, error) {
```

Comment:
Depending on user it might be useful create KubernetesClient from config or existing client. I expect Kubernetes will prefer to use existing client as it would want avoid duplicating etcd connection. The downside of this flexibility